### ISSUE TITLE
[REQUEST] Make docker build action faster

### ISSUE THREAD

**Issue by kingbri1** [theroyallab/tabbyAPI#186]
### Problem

The docker action doesn't cache anything when building. This causes the build times to be over 10 minutes for every push.

### Solution

Use an action to cache the intermediate build steps.

### Explanation

Doing this makes the image build build faster. This most likely requires a third-party build step.

### Acknowledgements

- [X] I have looked for similar requests before submitting this one.
- [X] I understand that the developers have lives and my issue will be answered when possible.
- [X] I understand the developers of this program are human, and I will make my requests politely.

**Comment by SecretiveShell** [comment]
one of the biggest things slowing the build is the fact that the .dockerignore file has been moved to the docker folder, meaning that the entire git history and a bunch of other stuff not needed is copied into the container. running git mv on the file would probably cut off a minute or two from the build and reduce the overall file size

**Comment by bdashore3** [comment]
Yep, merged that in from your PR. But the most important thing to significantly cut down on build times is to have a cache of some sort with actions. There's probably something out there.

**Comment by SecretiveShell** [comment]
https://docs.docker.com/build/cache/backends/gha/ seems to be a configuration option for the system we already use. @AmgadHasan can you provide any insights on this?

**Comment by AmgadHasan** [comment]
> https://docs.docker.com/build/cache/backends/gha/ seems to be a configuration option for the system we already use. @AmgadHasan can you provide any insights on this?

There is a cache size limit of 10 GB per repo. TabbyAPI image is ~ 5GB in size so the cache limit will be reached quickly.

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy

**Comment by kingbri1** [comment]
Added a concurrency group to run one docker action at a time, so speed issues aren't much of a problem anymore.
